submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,429 @@
+                    }
+                    if (
+                        !mutation.target.classList.contains("visible") &&
+                        SELECT_CONTAINER.getAttribute("aria-expanded")
+                    ) {
+                        SELECT_CONTAINER.removeAttribute("aria-expanded");
+                    }
+                }
+            }
+        };
+    const observer = new MutationObserver(callback);
+    observer.observe(OPTIONS_CONTAINER, options);
+}
+
+observeOptionsListVisibility();
+
+SELECT_CONTAINER.addEventListener("focusout", (e) => {
+    if (
+        checkOptionsVisibility() &&
+        e.relatedTarget !== null &&
+        !e.relatedTarget.classList.contains("list__value") &&
+        !e.relatedTarget.classList.contains("select__container")
+    )
+        OPTIONS_CONTAINER.classList.toggle("visible");
+});
+
+document.addEventListener("keydown", (keyEvent) => {
+    if (keyEvent.target === SELECT_CONTAINER) {
+        if (keyEvent.code === "Space" || keyEvent.code === "Enter") {
+            keyEvent.preventDefault();
+            if (!checkOptionsVisibility()) {
+                OPTIONS_CONTAINER.classList.add("visible");
+                if (isSelectModified()) {
+                    higlightSelectedOption(getSelectOption(selectedOption));
+                } else {
+                    selectedOption = OPTIONS_LIST[0].textContent;
+                    higlightSelectedOption(OPTIONS_LIST[0]);
+                }
+            } else {
+                updateCards(keyEvent);
+                OPTIONS_CONTAINER.classList.remove("visible");
+            }
+        }
+    }
+    if (keyEvent.code === "Escape" && checkOptionsVisibility()) {
+        OPTIONS_CONTAINER.classList.remove("visible");
+    }
+    if (
+        checkOptionsVisibility() &&
+        (keyEvent.code == "ArrowUp" || keyEvent.code == "ArrowDown")
+    ) {
+        keyEvent.preventDefault();
+        focusOnItem(keyEvent.code);
+    }
+    resetPreviouslySelectedOptions(selectedOption);
+});
+
+document.querySelector("#showFiltersButton").addEventListener("click", (e) => {
+    FILTERS_CONTAINER.classList.toggle("display");
+});
+
+document.addEventListener("click", (e) => {
+    if (checkOptionsVisibility()) {
+        if (e.target != OPTIONS_CONTAINER && e.target != SELECT_CONTAINER) {
+            OPTIONS_CONTAINER.classList.toggle("visible");
+        }
+    }
+});
+
+document.querySelector("#sort").addEventListener("click", (e) => {
+    if (
+        e.target.classList.contains("list__label") ||
+        e.target.classList.contains("select__container")
+    ) {
+        selectedOption = !SELECT_CONTAINER.getAttribute("value")  ? OPTIONS_LIST[0].textContent : e.target.textContent;
+        OPTIONS_CONTAINER.classList.toggle("visible");
+    }
+});
+
+document.querySelector("#search").addEventListener("input", (e) => {
+    const inputString = e.target.value,
+        filteredArray = findMatchesWithPropertiesValues(
```

Comment:
What motivation behind this function? 🤔

Reply:
This one was needed to handle an event when the user starts to type something in the search input to perform search filtering :eyes: